packages/syft/src/syft/node/credentials.py

```diff
@@ -53,10 +54,9 @@ def __hash__(self) -> int:
 class SyftSigningKey(SyftBaseModel):
     signing_key: SigningKey
 
-    def __init__(self, signing_key: Union[str, SigningKey]) -> None:
-        if isinstance(signing_key, str):
-            signing_key = SigningKey(bytes.fromhex(signing_key))
-        super().__init__(signing_key=signing_key)
+    @pydantic.validator("signing_key", pre=True, always=True)
+    def make_verify_key(cls, v: Union[str, SigningKey]) -> SigningKey:
+        return SigningKey(bytes.fromhex(v)) if isinstance(v, str) else v
```

Comment:
Q: Should we also throw an error, if any other value other than SigningKey or str is passed?

Reply:
Pydantic automatically throws an ValueError if value is not SigningKey.